ga4gh/datamodel/reads.py

```diff
@@ -91,6 +91,124 @@ def setFlag(flagAttr, flag):
         return flagAttr | flag
 
 
+class AlignmentDataMixin(datamodel.PysamDatamodelMixin):
+    """
+    Mixin class that provides methods for getting read alignments
+    from bam files
+    """
+    def _getReadAlignments(
+            self, reference, start=None, end=None, readGroup=None):
+        """
+        Returns an iterator over the specified reads
+        """
+        # TODO If reference is None, return against all references,
+        # including unmapped reads.
+        samFile = self.getFileHandle(self._samFilePath)
+        referenceName = reference.getLocalId().encode()
+        # TODO deal with errors from htslib
+        start, end = self.sanitizeAlignmentFileFetch(start, end)
+        readAlignments = samFile.fetch(referenceName, start, end)
+        for readAlignment in readAlignments:
+            tags = dict(readAlignment.tags)
+            if readGroup is None:
+                if 'RG' in tags:
+                    alignmentReadGroupId = tags['RG']
+                    alignmentReadGroup = self._readGroupLocalIdMap[
+                        alignmentReadGroupId]
+                    yield self.convertReadAlignment(
+                        readAlignment, alignmentReadGroup)
+                else:
+                    # TODO what to do for unmapped reads (i.e. no
+                    # alignment read group context)? skip for now
+                    pass
```

Comment:
Er no, hold on. Is being unmapped the same thing as not being in a ReadGroup? I thought being unmapped meant there was no reference and coordinates, and not belonging to a readGroup essentially means that the read doesn't contain an RG tag. If this is true we shouldn't conflate the two ideas.

Do we need the third part of the ID string? If we get ``datasetId:readGroupSetId:readGroupId:readId`` then the read belongs to a read group, and if we get ``datasetId:readGroupSetId:readId`` we know it doesn't? Seems simple enough to tell the two cases apart. I'm probably misunderstanding what's going though.